toolkit/tools/imagecustomizer/main.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package main
+
+import (
+	"log"
+	"os"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerlib"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app = kingpin.New("imagecustomizer", "Customizes a pre-built CBL-Mariner image")
+
+	buildDir          = app.Flag("build-dir", "Directory to run build out of.").Required().String()
+	imageFile         = app.Flag("image-file", "Path of the base CBL-Mariner image which the customization will be applied to.").Required().String()
+	outputImageFile   = app.Flag("output-image-file", "Path to write the customized image to.").Required().String()
+	outputImageFormat = app.Flag("output-image-format", "Format of output image. Supported: vhd, vhdx, qcow2, raw.").Required().Enum("vhd", "vhdx", "qcow2", "raw")
+	configFile        = app.Flag("config-file", "Path of the image customization config file.").Required().String()
+	logFile           = exe.LogFileFlag(app)
+	logLevel          = exe.LogLevelFlag(app)
+)
+
+func main() {
+	kingpin.MustParse(app.Parse(os.Args[1:]))
```

Comment:
nit: I think we are hoping to have profiling and timestamping as part of our core set of boilerplate bits. Not critical here but something to keep in mind for later versions. (Actually, now that I look I see `boilerplate.go` doesn't actually have the profiling stuff. I'll see about adding that later)

Reply:
From the new precaher.go tool I wrote:
```golang
	profFlags     = exe.SetupProfileFlags(app)
	timestampFile = app.Flag("timestamp-file", "File that stores timestamps for this program.").String()
```
```golang
	prof, err := profile.StartProfiling(profFlags)
	if err != nil {
		logger.Log.Warnf("Could not start profiling: %s", err)
	}
	defer prof.StopProfiler()

	timestamp.BeginTiming("precacher", *timestampFile)
	defer timestamp.CompleteTiming()

```

Reply:
Added.